### PR TITLE
feat: dashboards list sidebar with search & add new dashboard

### PR DESCRIPTION
**Pull request by roxanehe**
### Type of Change

- [ ] New Feature

### Description

1) a sidebar display a list of dashboards
2) a search & filter dashboard functionality,  based on keywords of names
3) a add new dashboards to the list functionality
4) storybook which shows 3 components and overall interactions

### Testing Instructions

1) 'npm run dev' to view the sidebar
2) 'npm run storybook' to see the storybook

### Screenshots

image
image
image

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -1,7 +1,15 @@
-<script setup lang="ts">
+<template>
+  <div class="app">
+    <Sidebar></Sidebar>
```
Comment (YoussefLaunchUp) [inline frontend/src/App.vue]: The Sidbebar component should not be in App. Keep App.vue as it was

**Review by YoussefLaunchUp — APPROVED** [review]
Small change requested. Great work!

File: frontend/src/App.vue (hunk 1 above)
Comment (roxanehe) [reply frontend/src/App.vue]: done ! App.vue is back to original

**Review by YoussefLaunchUp — APPROVED** [review]
Great work

**Review by MjxOro — COMMENTED** [review]
LGTM!

**Review by MjxOro — APPROVED** [review]
LGTM!

**Review by neli-rashkova — APPROVED** [review]
LGTM! Great work

**Pull request merged**